src/averageAges.js

```diff
@@ -20,6 +20,24 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const filterMenCallback = function(person) {
+    const isMan = person.sex === 'm';
+    const isSameCentury = Math.ceil(person.died / 100) === century;
+
+    return century ? isMan && isSameCentury : isMan;
+  };
+
+  function getAverageAge(peoples) {
+    return peoples
+      .map(person => person.died - person.born)
+      .reduce((prevAge, nextAge) => prevAge + nextAge, 0) / peoples.length;
+  }
+
+  const filteredMen = people.filter(person => filterMenCallback(person));
+
+  const calculateAge = getAverageAge(filteredMen);
+
+  return calculateAge;
 }
```

Comment:
```suggestion
//  Helpers
function getAverageValue(numbers) {
    return numbers.reduce((prevNumber, nextNumber) => {
      return prevNumber + nextNumber;
    }, 0) / numbers.length;
  }
  
function getAverageAge(peoples) {
    return getAverageValue(peoples.map(person => person.died - person.born))
}
  
// Task functions
  const filterMenCallback = function(person) {
    const isMan = person.sex === 'm';
    const isSameCentury = Math.ceil(person.died / 100) === century;

    return century ? isMan && isSameCentury : isMan;
  };

  const filteredMen = people.filter(person => filterMenCallback(person));

  const calculateAge = getAverageAge(filteredMen);

  return calculateAge;
}
```

Reply:
I DID NOT KNOW I CAN DO ANYTHING OUTSIDE THESE premade functions I did not know I am allowed  🤣